src/EFCore/Metadata/Internal/InternalSkipNavigationBuilder.cs

```diff
@@ -122,6 +122,7 @@ public virtual bool CanSetForeignKey([CanBeNull] ForeignKey foreignKey, Configur
             return (Metadata.DeclaringEntityType
                     == (Metadata.IsOnDependent ? foreignKey.DeclaringEntityType : foreignKey.PrincipalEntityType))
                             && (Metadata.Inverse?.AssociationEntityType == null
+                                || Metadata.Inverse?.AssociationEntityType?.IsAutomaticallyCreatedAssociationEntityType == true
```

Comment:
Can be simplified due to short-circuiting
```suggestion
                                || Metadata.Inverse.AssociationEntityType.IsAutomaticallyCreatedAssociationEntityType
```